app/components/reactions/reactions.js

```diff
@@ -44,8 +44,10 @@ export default class Reactions extends PureComponent {
     };
 
     componentDidMount() {
-        const {actions, postId} = this.props;
-        actions.getReactionsForPost(postId);
+        const {actions, postId, reactions} = this.props;
+        if (!reactions?.length) {
+            actions.getReactionsForPost(postId);
```

Comment:
Can you check this part of code once? When running locally i see that this is being called when i switch channel or when i add a first reaction to a message.

Reply:
@sudheerDev this should be fixed now.. the reactions is actually a `Map` and not an `Array` so we need to check for `size` instead of `length`